src/lib/pmd/PmdCatalogWrapper.ts

```diff
@@ -84,17 +84,64 @@ export class PmdCatalogWrapper extends PmdSupport {
     return JSON.parse(rawCatalog.toString());
   }
 
-  protected buildCommandArray(): [string, string[]] {
+  protected async buildCommandArray(): Promise<[string, string[]]> {
     const command = 'java';
     // NOTE: If we were going to run this command from the CLI directly, then we'd wrap the classpath in quotes, but this
     // is intended for child_process.spawn(), which freaks out if you do that.
-    const args = ['-cp', this.buildClasspath().join(':'), MAIN_CLASS, PMD_LIB, PMD_VERSION, SUPPORTED_LANGUAGES.join(',')];
+    const classpathEntries = await this.buildClasspath();
+    const parameters = await this.buildCatalogerParameters();
```

Comment:
Performance: These two lines of code could be executed at the same time if you replace them with `const [classpathEntries, parameters] = await Promise.all([this.buildClasspath(), this.buildCatalogerParameters()]);`.

Reply:
Good to know! Thanks.